lib/Serialization/Serialization.cpp

```diff
@@ -579,10 +579,11 @@ void Serializer::writeHeader(const SerializationOptions &options) {
     ModuleName.emit(ScratchRecord, M->getName().str());
 
     SmallString<32> versionStringBuf;
+    auto version = M->getASTContext().LangOpts.EffectiveLanguageVersion;
     llvm::raw_svector_ostream versionString(versionStringBuf);
-    versionString << Version::getCurrentLanguageVersion();
+    versionString << version;
```

Comment:
Then I'm .. perhaps even more confused here. If the short version is intended for diagnostic output -- of the form "sorry, this is a version <m.n> module!" then I would think it should be set to the effective language used to build that module, no? I.e. the way I have it set in this proposal.

I'm happy to change this if I'm misunderstanding, I'm just not sure what to change it to, or why.